xarray/core/indexing.py

```diff
@@ -345,20 +428,31 @@ def __init__(self, array, key=None):
             Array indexer. If provided, it is assumed to already be in
             canonical expanded form.
         """
-        if key is None:
-            key = (slice(None),) * array.ndim
-        self.array = array
-        self.key = key
+        # We need to avoid doubly wrapping.
+        if isinstance(array, type(self)):
+            self.array = array.array
+            self.key = array.key
+            if key is not None:
+                self.key = self._updated_key(key)
+
+        else:
+            if key is None:
+                key = (slice(None),) * array.ndim
+                if len(key) > 0:
+                    key = OuterIndexer(key)
```

Comment:
As your comment, some ArrayWrappers seem to check `type(key) is tuple`, if key is empty.
`to_tuple` method solves it.